File: 2054-two-best-non-overlapping-events/2054-two-best-non-overlapping-events.py
class Solution:
    def maxTwoEvents(self, events: List[List[int]]) -> int:
        res = 0
        maxVal = 0
        events = sorted(events)
        pq = []
        for event in events:
            while len(pq) > 0 and event[0] > pq[0][0]:
                maxVal = max(maxVal, pq[0][1])
                heapq.heappop(pq)
            heapq.heappush(pq, (event[1], event[2]))
            res = max(res, event[2] + maxVal)
        return res

File: 2054-two-best-non-overlapping-events/2054-two-best-non-overlapping-events.cpp
class Solution {
public:
    int maxTwoEvents(vector<vector<int>>& events) {
        int res = 0;
        int maxVal = 0;
        sort(events.begin(), events.end());
        priority_queue<pair<int, int>, vector<pair<int, int>>, greater<pair<int, int>>> pq;
        for (int i = 0; i < events.size(); i++) {
            while (!pq.empty() && events[i][0] > pq.top().first) {
                maxVal = max(maxVal, pq.top().second);
                pq.pop();
            }
            pq.push({events[i][1], events[i][2]});
            res = max(res, events[i][2] + maxVal);
        }
        return res;
    }
};